### PR TITLE
mgr/dashboard: service form hosts selection only show up to 10 entries

### PR DESCRIPTION
**Pull request by nmunet**
Fixes: https://tracker.ceph.com/issues/67413

Signed-off-by: Naman Munet <nmunet@redhat.com>

We have exposed UIRouter API to fetch all hosts information!
image
image
image

https://github.com/user-attachments/assets/19dad62b-5761-453c-9e5b-7a4f2e6faf74
form with dummy values

Screenshot from 2024-09-09 21-57-16

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [x] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by nmunet** [comment]
jenkins test make check

**Review by Pegonzal — CHANGES_REQUESTED** [review]
Hey @nmunet thanks for the work with this!

However I think we should look for another approach, for two reasons:

- The `minimal_health` endpoint should stay, as the name implies, minimal. Feeding it more data is to go against it's original purpose
- The original issue comes from the host pagination, which the `minimal_health` endpoint lacks. This pagination is in place to save on load times by retrieving only the needed information, some endpoints might manage a lot of data (some clusters are huge) and the host list is one of them, so we have to be careful when processing this data.

**Comment by nmunet** [comment]
jenkins test api

File: src/pybind/mgr/dashboard/controllers/host.py [hunk 1]
```diff
@@ -512,3 +512,10 @@ def labels(self) -> List[str]:
     @handle_orchestrator_error('host')
     def inventory(self, refresh=None):
         return get_inventories(None, refresh)
+
+    @Endpoint('GET')
+    @ReadPermission
+    @raise_if_no_orchestrator([OrchFeature.HOST_LIST])
+    @handle_orchestrator_error('host')
+    def all_hosts(self):
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/controllers/host.py:520]: and also add a description about this method on why its needed.
```suggestion
    def list(self):
```

**Review by nizamial09 — APPROVED** [review]
lgtm, thank you @nmunet. btw there's 3 different commit serving the same purpose. let's squash them all into one.

**Review by nizamial09 — APPROVED** [review]
Thanks for addressing the comments and taking care of the nvmeof hosts selection as well.

**Comment by nizamial09** [comment]
e2e failures are unrelated

**Review by Pegonzal — APPROVED** [review]

**Pull request merged**